tests/lib/Creator/DataSourceTestItemCreator.php

```diff
@@ -0,0 +1,223 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzRecommendationClient\Tests\Creator;
+
+use ArrayIterator;
+use EzSystems\EzRecommendationClient\Tests\Stubs\Criteria;
+use EzSystems\EzRecommendationClient\Tests\Stubs\Item;
+use EzSystems\EzRecommendationClient\Tests\Stubs\ItemList;
+use EzSystems\EzRecommendationClient\Tests\Stubs\ItemType;
+use Ibexa\Contracts\Personalization\Criteria\CriteriaInterface;
+use Ibexa\Contracts\Personalization\Value\ItemInterface;
+use Ibexa\Contracts\Personalization\Value\ItemListInterface;
+use Ibexa\Contracts\Personalization\Value\ItemTypeInterface;
+use Traversable;
+
+final class DataSourceTestItemCreator
+{
+    /**
+     * @phpstan-param ?iterable<string, array{
+     *  'item_type_identifier': string,
+     *  'item_type_name': string,
+     *  'languages': array<string>,
+     *  'limit': int,
+     * }> $itemsConfig
+     *
+     * @return Traversable<\Ibexa\Contracts\Personalization\Value\ItemInterface>
+     */
+    public static function createTestItems(?iterable $itemsConfig = null): Traversable
+    {
+        return self::createTestItemsForConfig(
+            $itemsConfig ?? self::getDefaultItemsConfig()
+        );
+    }
+
+    public static function createTestItem(
+        int $counter,
+        string $itemId,
+        string $itemTypeIdentifier,
+        string $itemTypeName,
+        string $language
+    ): ItemInterface {
+        return new Item(
+            $itemId,
+            self::createTestItemType($itemTypeIdentifier, $itemTypeName),
+            $language,
+            self::createTestItemAttributes(
+                $counter,
+                $itemTypeIdentifier,
+                $itemTypeName,
+                $language
+            )
+        );
+    }
```

Comment:
This can be converted or use internally a generator function (that will store the last used ID), which will remove the need for all the upstream methods to read the IDs or `count` returned elements.